File: 1991-find-the-middle-index-in-array/1991-find-the-middle-index-in-array.cpp
class Solution {
public:
    int findMiddleIndex(vector<int>& nums) {
        int n = nums.size();
        int left = 0;
        int sum = 0;

        if (n == 0) {
            return -1;
        }

        for (int i = 0; i < n; i++) {
            sum += nums[i];
        }

        for (int i = 0; i < n; i++) {
            sum -= nums[i];
            if (sum == left){
                return i;
            }
            left += nums[i];
        }
        return -1;
    }
};

File: 1991-find-the-middle-index-in-array/1991-find-the-middle-index-in-array.py
class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        n = len(nums)
        left = 0
        sum = 0
        
        if (n == 0):
            return -1

        for i in range(0, len(nums)):
            sum += nums[i]
        
        for i in range(0, len(nums)):
            sum -= nums[i]
            if sum == left:
                return i
            left += nums[i]
            
        return -1